internal/test/checks/checks.go

```diff
@@ -53,3 +56,103 @@ func ErrorIsNotf(t *testing.T, err, target error, format string, msg ...string)
 		t.Fatalf(format, msg)
 	}
 }
+
+type TestingT interface {
+	Fatalf(format string, args ...any)
+	Errorf(format string, args ...any)
+}
```

Comment:
Do we even need to abstract it as an interface, or can we use `*testing.T`?

Reply:
This works for both unittest and benchmark.